From 7184bdd9074361e206760ae4ee6a9a2c42bbd338 Mon Sep 17 00:00:00 2001
From: Thierry Vignaud <thierry.vignaud@gmail.com>
Date: Thu, 15 Oct 2015 19:30:48 +0200
Subject: [PATCH] fix linking with pangocairo

thus fixing this:

http://pkgsubmit.mageia.org/autobuild/cauldron/x86_64/core/2015-10-13/perl-Pango-1.226.0-1.mga6.src.rpm/build.0.20151013150513.log

(...)
xs/PangoCairo.xs: In function 'XS_Pango__Cairo__FontMap_create_context':
xs/PangoCairo.xs:85:2: warning: 'pango_cairo_font_map_create_context' is deprecated: Use 'pango_font_map_create_context' instead [-Wdeprecated-declarations]
  context = pango_cairo_font_map_create_context (fontmap);
  ^
In file included from ./pango-perl.h:28:0,
                 from xs/PangoCairo.xs:19:
/usr/include/pango-1.0/pango/pangocairo.h:93:15: note: declared here
 PangoContext *pango_cairo_font_map_create_context (PangoCairoFontMap *fontmap);
               ^
[ CC xs/PangoLayout.c ]
[ CC xs/PangoAttributes.c ]
rm -f blib/arch/auto/Pango/Pango.so
[ LD blib/arch/auto/Pango/Pango.so ]
chmod 755 blib/arch/auto/Pango/Pango.so
"/usr/bin/perl5.22.0" -MExtUtils::Command::MM -e 'cp_nonempty' -- Pango.bs blib/arch/auto/Pango/Pango.bs 644
Generating POD...
/usr/bin/perl5.22.0: symbol lookup error: blib/arch/auto/Pango/Pango.so: undefined symbol: pango_cairo_font_map_get_type
Makefile:1408: recipe for target 'build/podindex' failed
make: *** [build/podindex] Error 127
error: Bad exit status from /home/pterjan/rpmbuild/tmp/rpm-tmp.MZ04QX (%build)
---
 Makefile.PL | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index e644209..0b3b910 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -161,7 +161,7 @@ my $libs = $pango_pkgcfg{libs};
 if ($have_cairo) {
 	push @deps, qw/Cairo/;
 	$inc .= $pangocairo_pkgcfg{cflags};
-	$libs .= $pangocairo_pkgcfg{libs};
+	$libs .= " " . $pangocairo_pkgcfg{libs};
 }
 
 my @typemaps = map { File::Spec->catfile(cwd(), $_) }
-- 
2.6.1

